docs/graphql/manual/hasura-cli/reference.rst

```diff
@@ -0,0 +1,85 @@
+.. meta::
+   :description: Haura GarphQL CLI configuration reference 
+   :keywords: hasura, docs, CLI, flags, config 
+
+Hasura CLI configuration reference 
+==================================
+
+.. contents:: Table of contents
+  :backlinks: none
+  :depth: 1
+  :local:
+
+Hasura CLI commands get key values from flags, ENV variables, ``.env`` file and configuration file ``config.yaml``. 
+
+Configuration file format
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The configuration file, ``config.yaml`` format with default values is given below.
+
+.. code-block:: none
+
+  version: 1
+  endpoint: http://localhost:8080
+  admin_secret:
+  access_key: #deprecated
+  api_paths:
+    query: v1/query
+    graphql: v1/graphql
+    config: v1alpha1/config
+    pg_dump: v1alpha1/pg_dump
+    version: v1/version
+  metadata_directory:
+  migrations_directory: migrations
+  actions:
+    kind: synchronous
+    handler_webhook_baseurl: http://localhost:3000
+    codegen:
+      framework:
+      output_dir:
+      uri:
+
+
+CLI command flags and environment variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. list-table::
+   :header-rows: 1
+   :widths: 10 20 10 25
+
+   * - Flag 
+     - ENV variable
+     - Configuration file key
+     - Description
+    
+   * - ``--endpoint``
+     - ``HASURA_GRAPHQL_ENDPOINT``
+     - ``endpoint``
+     - http(s) endpoint for Hasura GraphQL Engine.
+
+   * - ``--admin-secret``
+     - ``HASURA_GRAPHQL_ADMIN_SECRET``
+     - ``admin_secret``  
+     - Admin secret for Hasura GraphQL Engine. 
+
+   * - ``--access-key``
+     - ``HASURA_GRAPHQL_ACCESS_KEY``
+     - ``access_key``
+     - Access key for Hasura GraphQL Engine. Note: Deprecated. Use admin 
+       secret instead. 
+
+   * - ``--kind``
+     - ``HASURA_GRAPHQL_ACTIONS_KIND``
+     - ``kind``
+     - Kind to use in action. Used in ``hasura actions create`` command. 
+
+   * - ``--webhook``
+     - ``HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL``
+     - ``handler_webhook_baseurl``
+     - Webhook to use in action. Used in ``hasura actions create`` command.
+
```

Comment:
All the configuration fields can be accessed via env.

Example:
`migrations_directory` can be accessed using `HASURA_GRAPHQL_MIGRATIONS_DIRECTORY`

It would be better if we can mention all the env variables that can be used with CLI.

Reply:
I included all the ENV vars and removed the flags column as the flag name changes based on the command. eg - `actions.kind` has the flag `--action-kind` for `hasura init` command and `--kind` for `hasura actions create`.

Reply:
@shahidhk Is it okay to have two different flags for action kind? or shall we update `actions create --kind` to `hasura actions create --action-kind`?

Reply:
`init` command should not be taking the actions kind flag. It should only take endpoint and admin secret. Everything else can be added after init.